### ISSUE TITLE
Conda package for SoFiA-2

### ISSUE THREAD

**Issue by sebastian-luna-valero** [SoFiA-Admin/SoFiA-2#24]
Hi,

I am interested in packaging SoFiA-2 into `conda-forge`:

https://github.com/conda-forge/staged-recipes/pull/13688

Would you like to be listed as a maintainer for the conda recipe?

Best regards,
Sebastian

**Comment by sebastian-luna-valero** [comment]
Hi,

The SoFiA-2 package has now been added to the `conda-forge` repository:
https://anaconda.org/conda-forge/sofia-2

If you are not familiar with conda, here is how to use it:
```
mkdir workdir
cd workdir
curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh -b -p conda-install
source conda-install/etc/profile.d/conda.sh 
conda create -n sofia-2 -c conda-forge sofia-2
conda activate sofia-2
sofia <sofia.par>
```
I hope that's helpful.

Best regards,
Sebastian

**Comment by SoFiA-Admin** [comment]
@sebastian-luna-valero: Thank you for the notification. I am pleased to hear that you have managed to create a conda package for SoFiA 2. As I have never used conda before, I don’t think it would make much sense for me to be listed as a maintainer, but thanks for the offer.

**Comment by sebastian-luna-valero** [comment]
No worries, thank you!

**Comment by vdhulst** [comment]
@sebastian-luna-valero: I tried the SoFiA conda package on a Scientific Linux system, CentOS 7 (<uname -a> gives: 3.10.0-1160.el7.x86_64 #1 SMP Wed Sep 30 08:53:05 CDT 2020 x86_64 x86_64 x86_64 GNU/Linux), but there it would not install. The <conda -c install conda-forge sofia-2> command would collect the package metadata, start an investigation of the existing environment and then just hang and eventually end with the message <\killed>. 
Is perhaps the problem that the distribution you made is only for Ubuntu, i.e. Debian kernels?
I also noticed that there is no conda distribution for MacOS.

**Comment by sebastian-luna-valero** [comment]
@vdhulst did you try with these specific steps?
```
mkdir workdir
cd workdir
curl -O https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh
bash Miniconda3-latest-Linux-x86_64.sh -b -p conda-install
source conda-install/etc/profile.d/conda.sh 
conda create -n sofia-2 -c conda-forge sofia-2
conda activate sofia-2
sofia <sofia.par>
```

**Comment by vdhulst** [comment]
@sebastian-luna-valero: no, but I tried just now and when I follow the above steps it works (N.B.this is more elaborate than what is advertised at https://anaconda.org/conda-forge/sofia-2) and requires running a bash shell (we use tcsh as our default shell). 

Conda does install many packages (some 32 in my case), including SoFiA-2 and WCSLIB, requiring some 620 MB in space.  Each time when one runs SoFiA one needs to start the sofia-2 environment. I find it a lot easier to just compile sofia-2 in the environment that already exists which has all dependencies satisfied and be able to run sofia-2 without having to start a special environment. 

But I can imagine some find it easier to do it this way. Maybe I am missing some essential elements/knowledge for working with conda.

**Comment by sebastian-luna-valero** [comment]
@vdhulst many thanks for your feedback.

The installation steps at https://anaconda.org/conda-forge/sofia-2 are automatically generated and I agree they are not helpful sometimes.

Indeed Conda consumes a lot of disk space. In general Conda is not perfect, it is just convenient and offers an alternative way to work with isolated environments. Imagine for example that a new version of SoFiA-2 adds a specific dependency with WCSLIB or any other package that conflicts with the existing ones. You could happily install and work the new version of SoFiA-2 in a separate conda environment, and switch between two without effort.

In summary, the conda installation just offers an alternative way of distributing SoFiA-2 and it is not intended to replace the original installation guide.

I hope that helps!
